Extension/src/Debugger/configurationProvider.ts

```diff
@@ -259,129 +350,36 @@ export class CppConfigurationProvider implements vscode.DebugConfigurationProvid
             });
         }));
         configs.push(defaultConfig);
+        // Sort tasks.
         return configs;
     }
 
-    /**
-	 * Error checks the provided 'config' without any variables substituted.
-	 */
-    resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration> {
-        // [Microsoft/vscode#54213] If config or type is not specified, return null to trigger VS Code to call provideDebugConfigurations
-        if (!config || !config.type) {
-            return null;
-        }
-
-        if (config.type === 'cppvsdbg') {
-            // Handle legacy 'externalConsole' bool and convert to console: "externalTerminal"
-            if (config.hasOwnProperty("externalConsole")) {
-                logger.getOutputChannelLogger().showWarningMessage(localize("debugger.deprecated.config", "The key '{0}' is deprecated. Please use '{1}' instead.", "externalConsole", "console"));
-                if (config.externalConsole && !config.console) {
-                    config.console = "externalTerminal";
-                }
-                delete config.externalConsole;
-            }
-
-            // Fail if cppvsdbg type is running on non-Windows
-            if (os.platform() !== 'win32') {
-                logger.getOutputChannelLogger().showWarningMessage(localize("debugger.not.available", "Debugger of type: '{0}' is only available on Windows. Use type: '{1}' on the current OS platform.", "cppvsdbg", "cppdbg"));
-                return undefined; // Stop debugging
-            }
+    private async loadDetectedTasks(): Promise<void> {
+        if (!DebugConfigurationProvider.detectedBuildTasks || DebugConfigurationProvider.detectedBuildTasks.length === 0) {
```

Comment:
If the user installs a new compiler while VS Code is still open, we might need to refresh this list.